corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -141,42 +160,91 @@ hqDefine("geospatial/js/geospatial_map", [
                 currCase.marker = marker;
             };
 
-            function moveMarkerToClickedCoordinate(coordinates) {
-                if (clickedMarker != null) {
-                    clickedMarker.remove();
-                }
-                if (draw.getMode() === 'draw_polygon') {
-                    // It's weird moving the marker around with the ploygon
-                    return;
-                }
-                clickedMarker = new mapboxgl.Marker({color: "FF0000", draggable: true});
-                clickedMarker.setLngLat(coordinates);
-                clickedMarker.addTo(map);
-            }
-
             // Handle click events here
             map.on('click', (event) => {
                 let coordinates = getCoordinates(event);
             });
             return self;
         };
 
+        var exportGeoJson = function(drawInstance) {
+            // Credit to https://gist.github.com/danswick/36796153bd86ce982a59043cbe0ac8f7
+            var exportButton = $("#btnExport");
+            var data = drawInstance.getAll();
+
+            if (data.features.length) {
+                // Stringify the GeoJson
+                var convertedData = 'text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(data));
+
+                // Create export
+                exportButton.attr('href', 'data:' + convertedData);
+                exportButton.attr('download','data.geojson');
+            }
+        };
+
+        var mapControlsModel = function () {
+            'use strict';
+            var self = {};
+            var mapboxinstance = map.getMapboxInstance();
+            var polygons = [];
```

Comment:
This var can be modelled as a ko.observable which should make it easier.

Reply:
See [this comment](https://github.com/dimagi/commcare-hq/pull/33062#discussion_r1219634865)